src/kernel.cpp

```diff
@@ -152,42 +42,14 @@ cl_ulong cvk_kernel::local_mem_size() const {
 }
 
 cl_int cvk_kernel::init() {
-    // Get a pointer to the arguments from the program
-    auto args = m_program->args_for_kernel(m_name);
-
-    if (args == nullptr) {
-        cvk_error("Kernel %s doesn't exist in program", m_name.c_str());
-        return CL_INVALID_KERNEL_NAME;
-    }
-
-    // Store a sorted copy of the arguments
-    m_args = *args;
-    std::sort(
-        m_args.begin(), m_args.end(),
-        [](kernel_argument a, kernel_argument b) { return a.pos < b.pos; });
-
-    // Create Descriptor Sets Layout
-    VkResult res;
-    auto vkdev = m_context->device()->vulkan_device();
-
-    std::unordered_map<VkDescriptorType, uint32_t> bindingTypes;
-    if (!build_descriptor_sets_layout_bindings_for_literal_samplers(
-            vkdev, bindingTypes)) {
-        return CL_INVALID_VALUE;
-    }
-
-    uint32_t num_resources;
-    if (!build_descriptor_sets_layout_bindings_for_arguments(
-            vkdev, bindingTypes, num_resources)) {
-        return CL_INVALID_VALUE;
+    cl_int errcode;
+    m_entry_point = m_program->get_entry_point(m_name, &errcode);
```

Comment:
I _think_ this is the first non-CL function that returns the value and sets an error code via a pointer passed as a parameter instead of doing the opposite. I don't think I feel too strongly about that though.

Reply:
`cvk_buffer::create()` does this. I can't actually see any non-CL functions that return an error code and create something as an output parameter (though there are functions that do not create stuff like `cvk_kernel::set_arg`). I also don't feel strongly about this though :-)

Reply:
Yeah, that one is almost a CL function :). I'm sure there are but let's not spend time on this.